### PR TITLE
Feature/adf 1196/sorting

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/ADF-1196

# Goal

Forward sorting instructions from FE to BE in order to apply proper ordering in ElasticSearch.

### CODE REVIEW

File: models/classes/search/SearchQuery.php [hunk 1]
```diff
@@ -99,4 +115,14 @@ public function getPage(): ?int
...
+    public function getSortBy(): string
+    {
+        return $this->sortBy;
+    }
+
+    public function getSortOrder(): string
+    {
+        return $this->sortOrder;
+    }
 }
```
Comment (shpran) [inline models/classes/search/SearchQuery.php:128]:
```suggestion
}

```

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: models/classes/auth/AbstractCredentials.php [hunk 2]
```diff
@@ -45,21 +45,18 @@ abstract public function getProperties();
...
-    protected function validate($properties)
+    protected function validate()
     {
-        $validatedProperties = array_keys($this->getProperties());
-        foreach ($properties as $key => $value) {
-            if (!in_array($key, $validatedProperties, false)) {
-                throw new \common_exception_ValidationFailed($key);
-            }
+        $extraProperties = array_diff_key($this->properties, $this->getProperties());
+        if ($extraProperties) {
```
[the remark this answers is not in the document]
Comment (gabrielfs7) [reply models/classes/auth/AbstractCredentials.php:58]: this commit does not belong to my PR, there was a git issue and it got here for some reason. It is already in the developer branch

File: models/classes/search/SearchQuery.php (hunk 1 above)
Comment (gabrielfs7) [reply models/classes/search/SearchQuery.php:128]: done

**Review by hectoras — APPROVED** [review]
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Target branch is correct
- [x] Commits are following conventional commits
- [x] New code is covered by tests
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Kitchen links are not exposed in the description

**Pull request merged**